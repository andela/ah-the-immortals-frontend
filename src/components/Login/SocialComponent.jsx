import React from 'react';
import PropTypes from 'prop-types';

const SocialComponent = ({ facebook, google, twitter }) => {
  return (
    <div>
      <button className="btn btn-link" id="fb" onClick={facebook} type="button">
        <img src="https://img.icons8.com/color/48/000000/facebook-circled.png" alt="avatar" />
      </button>
      <button className="btn btn-link" id="ggl" onClick={google} type="button">
        <img src="https://img.icons8.com/color/48/000000/google-plus.png" alt="avatar" />
      </button>
      <button className="btn btn-link" id="twt" onClick={twitter} type="button">
        <img src="https://img.icons8.com/color/48/000000/twitter-circled.png" alt="avatar" />
      </button>
    </div>
  );
};

SocialComponent.propTypes = {
  facebook: PropTypes.func.isRequired,
  google: PropTypes.func.isRequired,
  twitter: PropTypes.func.isRequired,
};

export default SocialComponent;
