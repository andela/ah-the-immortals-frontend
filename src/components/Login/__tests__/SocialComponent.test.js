import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SocialComponent from '../SocialComponent';

describe('<SocialComponent/>', () => {
  it('renders social component', () => {
    const mockfacebook = jest.fn();
    const mockgoogle = jest.fn();
    const mocktwitter = jest.fn();
    const wrapper = mount(
      <SocialComponent
        facebook={mockfacebook}
        google={mockgoogle}
        twitter={mocktwitter}
      />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
