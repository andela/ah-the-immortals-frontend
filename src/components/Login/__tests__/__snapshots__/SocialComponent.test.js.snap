// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialComponent/> renders social component 1`] = `
<SocialComponent
  facebook={[MockFunction]}
  google={[MockFunction]}
  twitter={[MockFunction]}
>
  <div>
    <button
      className="btn btn-link"
      id="fb"
      onClick={[MockFunction]}
      type="button"
    >
      <img
        alt="avatar"
        src="https://img.icons8.com/color/48/000000/facebook-circled.png"
      />
    </button>
    <button
      className="btn btn-link"
      id="ggl"
      onClick={[MockFunction]}
      type="button"
    >
      <img
        alt="avatar"
        src="https://img.icons8.com/color/48/000000/google-plus.png"
      />
    </button>
    <button
      className="btn btn-link"
      id="twt"
      onClick={[MockFunction]}
      type="button"
    >
      <img
        alt="avatar"
        src="https://img.icons8.com/color/48/000000/twitter-circled.png"
      />
    </button>
  </div>
</SocialComponent>
`;
